utils/clusterutils.go

```diff
@@ -30,6 +30,14 @@ import (
 var propagationPolicy = metav1.DeletePropagationForeground
 var terminationGracePeriodSeconds = int64(1)
 
+// Volume mount to copy the sleep binary into.
+// To allow the image puller to cache scratch images, an initContainer copies
+// the sleep binary to this volume mount. As a result, every container has
+// access to the sleep binary via this volume mount.
+var containerVolumeMounts = []corev1.VolumeMount{
+	{Name: "kip", MountPath: "/kip"},
+}
```

Comment:
Optional suggestion: move all volume name/mount path related stuff into consts and variables at the top rather than hard-coding strings down below:

```golang
const (
  kipVolumeName = "kip"
  kipVolumeMountPath = "/kip"
  copySleepCommand = "cp /bin/sleep /kip/sleep"
  containerSleepCommand = "/bin/sleep"
)

var (
  containerVolumeMounts = []corev1.VolumeMount{
    {
      Name: kipVolumeName,
      MountPath: kipVolumeMountPath,
    }
  }
)

// (...)
        InitContainers: []corev1.Container{{
            Name:            "copy-sleep",
            Image:           cfg.ImagePullerImage,
            ImagePullPolicy: corev1.PullAlways,
            Command:         []string{"/bin/sh"},
            Args:            []string{"-c", copySleepCommand},
            VolumeMounts:    containerVolumeMounts,
            Resources:       getContainerResources(cfg),
          }},
        Volumes:          []corev1.Volume{{Name: kipVolumeName}},

// (...)
  for name, image := range images {
    containers[idx] = corev1.Container{
      Name:            name,
      Image:           image,
      Command:         []string{containerSleepCommand},
      Args:            []string{"720h"},
      Resources:       cachedImageResources,
      Resources:       getContainerResources(cfg),
      ImagePullPolicy: corev1.PullAlways,
      VolumeMounts:    containerVolumeMounts,
    }
    idx++
  }
```

Reply:
Thank you, updated in my latest commit